com.unity.uiwidgets/Runtime/painting/text_painter.cs

```diff
@@ -29,21 +63,36 @@ public class TextPainter {
         string _ellipsis;
         float _lastMinWidth;
         float _lastMaxWidth;
+        TextWidthBasis _textWidthBasis;
+        TextHeightBehavior _textHeightBehavior;
 
-        public TextPainter(TextSpan text = null,
+        public TextPainter(InlineSpan text = null,
             TextAlign textAlign = TextAlign.left,
             TextDirection textDirection = TextDirection.ltr,
             float textScaleFactor = 1.0f,
             int? maxLines = null,
             string ellipsis = "",
-            StrutStyle strutStyle = null) {
+            StrutStyle strutStyle = null,
+            TextWidthBasis textWidthBasis = TextWidthBasis.parent,
+            TextHeightBehavior textHeightBehavior = null) {
+            D.assert(text == null || text.debugAssertIsValid());
+            D.assert(maxLines == null || maxLines > 0);
             _text = text;
             _textAlign = textAlign;
             _textDirection = textDirection;
             _textScaleFactor = textScaleFactor;
             _maxLines = maxLines;
             _ellipsis = ellipsis;
             _strutStyle = strutStyle;
+            _textWidthBasis = textWidthBasis;
+            _textHeightBehavior = textHeightBehavior;
+        }
+
+        void markNeedsLayout() {
+            _paragraph = null;
```

Comment:
_paragraph is a C# object which will hold a c++ object. I see no place to call _paragraph._dispose() in this class, is there any memory leak issue on the c++ side here?

Reply:
Paragraph implements NativeWrapper, deconstructor of which calls _dispose.